### DISCUSSION TITLE
Can I have folders inside an autogenerated directory with each folder being just a page in the sidebar?

### DISCUSSION THREAD

**Discussion by darrenparkinson** [withastro/starlight#2207] [Q&A] [answered]
Essentially I am getting folders of documentation from different locations. 

Each folder may have it's own assets folder etc. that the markdown file references with relative paths, so these could overlap.

In this instance, it would be ideal if I could take the "README.md" lets say of each of those folders and present it at the top level of the sidebar group it is in but with all the assets in tact.

For example, in the sidebar configuration, we have:

```mjs
{
    label: 'My Group of Things',
    autogenerate: { directory: 'things' },
},
```

And the directory tree looks like this:

```bash
src/content/docs/things
├── thing1
│   ├── README.md
│   └── assets
│       └── image1.png
└── thing2
    ├── README.md
    └── images
        └── image1.png
```

Rather than that look like this:

<img width="293" alt="image" src="https://github.com/user-attachments/assets/23a6cb34-43b7-461b-92c8-cad50555c351">

I'd like it to look like this:

<img width="291" alt="image" src="https://github.com/user-attachments/assets/47da69a1-8572-4197-8ac1-1769fb1fd54e">

Does anyone know if this is already possible, or what customisation I might need to do to achieve this?

Many thanks in advance.

**Comment by delucis** [comment, marked as answer]
Hi @darrenparkinson, you have two options here, or rather I guess two versions of the same option depending on the complexity of your project.

I’d recommend switching away from full autogeneration to reference pages with slugs, so for example based on your content tree, this could look like:

```js
{
    label: 'My Group of Things',
    items: ['thing1/readme', 'thing2/readme'],
},
```

Internal link shorthand like this still loads all the metadata like page title etc. from the content files, but they are specified directly giving you full control of their nesting, order etc.

Version two is kind of an update on this if you have a lot of instances of this and really do want to autogenerate rather than manually maintaining a list of all these. Instead of authoring the `items` manually, you can read your content directory to generate the list of these slugs automatically (for example with Node’s `fs.readdirSync()` or with a glob library). I can go into more detail if it would be helpful, but hopefully that’s a good hint in the right direction!

**Reply by darrenparkinson** [reply]
Ah yes perfect of course, that makes total sense. I’m already automating the contents elsewhere so actually that should be pretty straight forward. 

Essentially I’m downloading documentation from all our GitLab repos and consolidating it into a single documentation site. Currently it uses mdbook for the final site, and I’m currently using Go with templates to download the content and create it. 

I tried starlight before but it didn’t have mermaid diagrams which we use quite a lot — that problem also now appears to be solved, so I’ll definitely be giving it another try since I think it looks nicer. 

Thanks again for the response. Super helpful and timely.
